### PR TITLE
Improved ECAM Messages System

### PR DESCRIPTION
**Pull request by tyler58546**
**Summary of Changes**

* ECAM messages area has been completely redone
* Messages can now show on the right side of the ECAM
* Messages will now show in a consistent order
* Added many new ECAM messages
    * T.O. INHIBIT
    * LDG INHIBIT
    * APU BLEED
    * APU AVAIL
    * LDG LT
    * SPEED BRK
    * AUTO BRK LO
    * AUTO BRK MED
    * AUTO BRK MAX
* Added takeoff memo
    * Shows 2 mins after second engine start
    * Hides after takeoff thrust application
* Added landing memo
    * Shows below 2000 feet during approach phase
    * Hides after slowing down below 80kts
* Added support for failures to show on ECAM
* Added NAV TCAS FAULT failure when ADIRS are not aligned
* Added CAB PR EXCESS CAB ALT failure when cabin altitude exceeds 10000ft (can only be triggered by slewing to an extreme altitude)

**Screenshots (if necessary)**

### CODE REVIEW

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1130]: Use `for ... of` here.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1198]: Use `for ... of` here.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1200]: Use `for ... of` here.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1215]: Can be `const`.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1224]: Can be `const`.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1269]: Use `for ... of` here.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1274]: Can be `const`.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1279]: Use `for ... of` here.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1285]: Can be `const`.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1332]: Some variables can be turned into `const`s here.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1348]: Some variables can be turned into `const`s here.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1351]: Can be `const`.

**Review by Benjozork — COMMENTED** [review]
Great improvements !

**Comment by nabil74** [comment]
AESEOME CANT WAIT TILL THESE ARE MERGED !!!

 

Regards
Nabil Abaki

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js
Comment (tyler58546) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1269]: Need the `i` for `i === 1`

**Review by wpine215 — CHANGES_REQUESTED** [review]
Change "IR IN ALGIN" to "IR IN ALIGN" in A320_Neo_UpperECAM.js

**Review by wpine215 — CHANGES_REQUESTED** [review]
Received T.O. INHIBIT message during cold-and-dark setup at gate, with APU + APU bleed on, parking brake engaged, seat belts on, and no smoking auto (engines not started yet or in ignition).

**Comment by wpine215** [comment]
What are the conditions necessary for GPWS Fault to display?

**Comment by tyler58546** [comment]
@wpine215 
> Change "IR IN ALGIN" to "IR IN ALIGN" in A320_Neo_UpperECAM.js

> Received T.O. INHIBIT message during cold-and-dark setup at gate, with APU + APU bleed on, parking brake engaged, seat belts on, and no smoking auto (engines not started yet or in ignition).

Fixed.
> What are the conditions necessary for GPWS Fault to display?

It doesn't currently display. I was just using it as an example to show that multiple failures can show under the same category.

**Review by wpine215 — CHANGES_REQUESTED** [review]
"LDG LT" memo stays on even after landing lights are retracted following takeoff.
"TO INHIBIT" memo did not show up at all during takeoff roll. (However, LDG INHIBIT works fine on landing)
"FLAPS T.O." should only be marked as completed when the flaps are actually extended to takeoff position, instead of relying on the flap handle position. (Same for flaps full on landing)

**Comment by tyler58546** [comment]
I fixed T.O. INHIBIT not showing. It looks like there aren't any good simvars for flap position and landing lights.

**Comment by griffon55** [comment]
Landing memo stay on after landing even on the taxiway.
Seat belt and smoking sign button not working now.

**Comment by AdenFlorian** [comment]
> Seat belt and smoking sign button not working now.
that fix just went to master, caused by different pr

**Pull request merged**
